Mathlib/Data/Multiset/DershowitzManna.lean

```diff
@@ -0,0 +1,409 @@
+/-
+Copyright (c) 2024 Haitian Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Haitian Wang, Malvin Gattinger
+-/
+import Mathlib.Tactic.Linarith
+import Mathlib.Data.Multiset.Basic
+import Mathlib.Logic.Relation
+
+/-!
+# Dershowitz-Manna ordering
+
+In this file we define the _Dershowitz-Manna ordering_ on multisets.
+We prove that, given a well-founded partial order on the underlying set,
+the Dershowitz-Manna ordering defined over multisets is also well-founded.
+
+## Main results
+
+- `MultisetDMLT` :  the standard definition fo the `Dershowitz-Manna ordering`.
+- `dm_wf` :       the main theorem about the `Dershowitz-Manna ordering` being well-founded.
+- `TransLT_eq_DMLT` : two definitions of the `Dershowitz-Manna ordering` are equivalent.
+
+## References
+
+* [Wikipedia, Dershowitz–Manna ordering*]
+(https://en.wikipedia.org/wiki/Dershowitz%E2%80%93Manna_ordering)
+
+* [CoLoR](https://github.com/fblanqui/color), a Coq library on rewriting theory and termination.
+  Our code here is inspired by their formalization and the theorem is called `mOrd_wf` in the file
+  [MultisetList.v](https://github.com/fblanqui/color/blob/1.8.5/Util/Multiset/MultisetOrder.v).
+
+-/
+
+variable {α : Type*}
+
+/-- The standard Dershowitz–Manna ordering: -/
+inductive MultisetDMLT [DecidableEq α] [Preorder α] (M N : Multiset α) : Prop :=
+  | DMLT : ∀ (X Y Z : Multiset α),
+    Y ≠ ∅ →
+    M = Z + X →
+    N = Z + Y →
+    (∀ x ∈ X, ∃ y ∈ Y, x < y) → MultisetDMLT M N
+
+/-- MultisetRedLT is a special case of MultisetDMLT. The transitive closure of it is used to define
+    an equivalent (proved later) version of the ordering. -/
+inductive MultisetRedLT [DecidableEq α] [LT α] (M N : Multiset α) : Prop :=
+  | RedLT : ∀ (X Y : Multiset α) (a : α) ,
+    (M = X + Y) →
+    (N = X + {a}) →
+    (∀ y, y ∈ Y → y < a) → MultisetRedLT M N
+
+open Relation
+
+/-- MultisetTransLT is the transitive closure of MultisetRedLT and is equivalent to MultisetDMLT
+    (proved later). -/
+def MultisetTransLT [DecidableEq α] [LT α] : Multiset α → Multiset α → Prop :=
```

Comment:
I'm not 100% sure exactly what the Mathlib convention is, but shouldn't the `Multiset` be a namespace instead of part of the name in CammelCase? i.e. `Multiset.transLT` and `Multiset.DMLT` above, etc?

Reply:
Sure! We have added the namespace 'Multiset' now;-)